transformer_engine/jax/sharding.py

```diff
@@ -54,34 +61,72 @@ class ShardingResource:
     tp_resource : str, default = None
         The axis name in Mesh used to split the hidden dimensions along.
         If it is None, then tensor parallelism is disabled.
+    fsdp_resource : str, default = None
+        The axis name in Mesh used to split the batch and weights along.
+        If it is None, then full-sharded data parallelism is disabled.
+    pp_resource : str, default = None
+        The axis name in Mesh used to split model layers. along.
+        If it is None, then pipeline parallelism is disabled.
     """
     dp_resource: str = None
     tp_resource: str = None
     fsdp_resource: str = None
+    pp_resource: str = None
 
 
-_GLOBAL_SHARD_RESOURCE = ShardingResource()
+_GLOBAL_MESH_RESOURCE = MeshResource()
 
 
 @contextmanager
-def global_shard_guard(resource: ShardingResource):
+def global_shard_guard(resource: MeshResource):
     """
-    A context manager to switch the global ShardingResource
+    A context manager to switch the global MeshResource
     """
-    global _GLOBAL_SHARD_RESOURCE
-    prev_gsr = _GLOBAL_SHARD_RESOURCE
+    global _GLOBAL_MESH_RESOURCE
+    prev_gmr = _GLOBAL_MESH_RESOURCE
     try:
-        _GLOBAL_SHARD_RESOURCE = resource
+        _GLOBAL_MESH_RESOURCE = resource
         yield
     finally:
-        _GLOBAL_SHARD_RESOURCE = prev_gsr
+        _GLOBAL_MESH_RESOURCE = prev_gmr
 
 
-def global_shard_resource() -> ShardingResource:
+def global_mesh_resource() -> MeshResource:
     """
-    A getter of  the global ShardingResource
+    A getter of  the global MeshResource
     """
-    return _GLOBAL_SHARD_RESOURCE
+    return _GLOBAL_MESH_RESOURCE
+
+
+def all_reduce_sum_along_dp_fsdp(x: jnp.array):
+    """
+    All-Reduce (Sum) along DP and FSDP mesh axes.
+    """
+    x = lax_paral_op(x, jax.lax.psum, global_mesh_resource().dp_resource)
+    return lax_paral_op(x, jax.lax.psum, global_mesh_resource().fsdp_resource)
+
+
+def all_reduce_sum_along_tp(x: jnp.array):
+    """
+    All-Reduce (Sum) along TP mesh axis.
+    """
+    return lax_paral_op(x, jax.lax.psum, global_mesh_resource().tp_resource)
+
+
+def all_reduce_max_along_all(x: jnp.array):
```

Comment:
The function isn't used. Do we need it?

Reply:
Remove for this PR, but will need in the further migration.